docs/spec/distribution/transactions.md

```diff
@@ -0,0 +1,396 @@
+# Transactions
+
+## TxWithdrawDelegation
+
+When a delegator wishes to withdraw their transaction fees it must send
+`TxWithdrawDelegation`. Note that parts of this transaction logic is also
+triggered each with any change in individual delegations, such as an unbond,
+redelegation, or delegation of additional tokens to a specific validator.  
+
+Generally, each time a withdrawal is made by a recipient the adjustment term
+must be modified for each block with a change in distributors shares since the
+time of last withdrawal.  This is accomplished by iterating over all relevant
+`PowerChange`'s stored in distribution state.
+
+
+```golang
+type TxWithdrawDelegation struct {
+    delegatorAddr sdk.AccAddress
```

Comment:
Do we want the ability to specify *which* delegation is delegator is withdrawing from?

They could have multiple delegations to separate validators (or even to the same validator), and might want to withdraw at separate times (or to avoid using too much gas).

Reply:
probably not something we want anytime in the near future - I _think_ that there are some shared computations when a delegator withdraws all of their tokens as opposed to doing this through multiple transactions.  - Due to how things are structured internally we would be able to introduce this down the road, but I think for now let's stick with the obviously required transactions of processing them all simultaneously.

Reply:
Makes sense :+1: